edi_voxel_sale_order_import/models/sale_order.py

```diff
@@ -0,0 +1,353 @@
+            if client_data.get("SupplierClientID"):
+                partner_data["ref"] = client_data.get("SupplierClientID")
+            else:
+                partner_data["ref"] = client_data.get("CustomerID")
+            partner = self._parse_partner_data_voxel(partner_data)
+            if partner:
+                vals.update(partner_id=partner.id)
+
+    def _parse_customers_data_voxel(self, vals, xml_root, error_msgs):
+        customer_elements = xml_root.xpath("//Customers/Customer")
+        if customer_elements:
+            customer_data = customer_elements[0].attrib
+            partner_data = self._get_partner_data_voxel(customer_data)
+            partner_data.update(name=customer_data.get("Customer"))
+            if customer_data.get("SupplierClientID"):
+                partner_data["ref"] = customer_data.get("SupplierClientID")
+            else:
+                partner_data["ref"] = customer_data.get("CustomerID")
+            partner = self._parse_partner_data_voxel(partner_data)
+            # Update partner_shipping_id value
+            if partner:
+                vals.update(partner_shipping_id=partner.id)
+            else:
+                # Add error message to error_msgs list
+                msg_fields = self._get_voxel_msg_fields("res.partner", partner_data)
+                error_msgs.append(
+                    _(
+                        "Couldn't find any <b>Delivery Address</b> "
+                        "corresponding to the following information extracted "
+                        "from the Voxel document:<br/>"
+                        "<ul>%s</ul>"
+                    )
+                    % (msg_fields)
+                )
+
+    def _parse_partner_data_voxel(self, data, raise_error=True):
+        domains = []
+        # fill domains list
+        if data.get("ref"):
+            domains.append([("ref", "=", data.get("ref"))])
+        partner = self.env["res.partner"].search(expression.AND(domains))
+        if len(partner) == 1:
+            return partner
+        elif len(partner) == 0:
+            domains = []
+        # If there are ZERO or more than ONE objects, continue filling domains
+        if data.get("country_id"):
+            domains.append([("country_id.code_alpha3", "=", data["country_id"])])
+        if data.get("state_id"):
+            domains.append([("state_id.name", "=", data["state_id"])])
+        if data.get("name"):
+            domains.append([("name", "=", data["name"])])
+        if data.get("email"):
+            domains.append([("email", "=", data["email"])])
+        if data.get("zip"):
+            domains.append([("zip", "=", data["zip"])])
+        elif data.get("city"):
+            domains.append([("city", "=", data["city"])])
+        partner = self.env["res.partner"].search(expression.AND(domains))
+        if len(partner) == 1:
+            return  # return the unique partner matching
```

Comment:
return partner # return the unique partner matching

Reply:
Done